core/src/splitting/SurvivalSplittingRule.cpp

```diff
@@ -0,0 +1,316 @@
+/*-------------------------------------------------------------------------------
+  This file is part of generalized random forest (grf).
+
+  grf is free software: you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation, either version 3 of the License, or
+  (at your option) any later version.
+
+  grf is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with grf. If not, see <http://www.gnu.org/licenses/>.
+ #-------------------------------------------------------------------------------*/
+
+#include <algorithm>
+
+#include "SurvivalSplittingRule.h"
+
+namespace grf {
+
+SurvivalSplittingRule::SurvivalSplittingRule(double alpha):
+    alpha(alpha) {
+}
+
+bool SurvivalSplittingRule::find_best_split(const Data& data,
+                                            size_t node,
+                                            const std::vector<size_t>& possible_split_vars,
+                                            const std::vector<double>& responses_by_sample,
+                                            const std::vector<std::vector<size_t>>& samples,
+                                            std::vector<size_t>& split_vars,
+                                            std::vector<double>& split_values,
+                                            std::vector<bool>& send_missing_left) {
+
+  double unused_test_value; // This output is only used in interal unit tests.
+  return find_best_split_internal(data, node, possible_split_vars, responses_by_sample,
```

Comment:
I have modified it to return the best logrank now, is it as you were thinking now?

Reply:
This works for me. I was thinking `find_best_split_internal` would still accept `split_vars`, `split_values`, and `send_missing_left`. Now it's a bit inconsistent, some `best_*` values are returned as parameters, while `best_logrank` is not. But I'm really getting into style minutiae here!

Reply:
Sorry, I read the that sentence as literally it should return something with `return`, I changed it to `void` and have it return everything as passed parameters as before. Now it is more consistent, but it still does not take the `split_vars` vectors, etc, since then it would only `best_logrank=0` outside, and initialize `best_value=0` etc inside, which would be a bit confusing?